includes/class-wc-payments-token-service.php

```diff
@@ -264,6 +265,21 @@ public function get_account_saved_payment_methods_list_item_sepa( $item, $paymen
 		return $item;
 	}
 
+	/**
+	 * Controls the output for Stripe Link on the My account page.
+	 *
+	 * @param  array                                        $item          Individual list item from woocommerce_saved_payment_methods_list.
+	 * @param  WC_Payment_Token|WC_Payment_Token_WCPay_Link $payment_token The payment token associated with this method entry.
+	 * @return array                                        Filtered item
+	 */
+	public function get_account_saved_payment_methods_list_item_link( $item, $payment_token ) {
+		if ( WC_Payment_Token_WCPay_Link::TYPE === strtolower( $payment_token->get_type() ) ) {
+			/* translators: %s is a registered Stripe Link email. */
+			$item['method']['brand'] = sprintf( esc_html__( 'Stripe Link email %s', 'woocommerce-payments' ), esc_html( $payment_token->get_email() ) );
```

Comment:
I wonder if [`sanitize_email()`](https://developer.wordpress.org/reference/functions/sanitize_email/) would be a better choice for the email address here. I don't think `esc_html()` will have much of an effect on email addresses whereas `sanitize_email()` is designed specifically to validate and sanitize email addresses. 

```suggestion
			$item['method']['brand'] = sprintf( esc_html__( 'Stripe Link email %s', 'woocommerce-payments' ), sanitize_email( $payment_token->get_email() ) );
```

Reply:
That's a good point, I didn't think of sanitizing the output because the email value should have been sanitized on the input ([here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payment-gateway-wcpay.php#L1411) I believe). 

So on one side, at first glance, I wouldn't expect sanitization here as sanitization is supposed to be done while saving the value. On the other side, sanitization probably shouldn't bring any data loss/corruption since the value was already sanitized before and thus can be used? But if used, maybe still apply `esc_html` on top of it, to be sure that eventual HTML tags are considered plain text.

Let me know what you think about all this! 👍

Reply:
Another thing I started to wonder right now is if with `esc_html` I'm not escaping it too early? Eventually, this value will be still escaped [in WooCommerce Core](https://github.com/Automattic/woocommerce.com/blob/trunk/plugins/woocommerce/templates/myaccount/payment-methods.php#L51), so my escaping here might be redundant. On top of that, I heard many times that the best way is to escape late (which in our case is WC Core which does the latest escape possible), to avoid the returned value updates somewhere in between.

Reply:
Speaking input sanitization: billing email is being saved upon order placement by the `/customer` API call, the data for which is [being prepared by the `WC_Payments_Customer_Service::map_customer_data`](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-customer-service.php#L311). 

The order is initially created by WooCommerce Core as [one of the first checkout processing steps.](https://github.com/Automattic/woocommerce.com/blob/trunk/plugins/woocommerce/includes/class-wc-checkout.php#L1256C5-L1256C5). 

The email field is being set in the order [here](https://github.com/Automattic/woocommerce.com/blob/trunk/plugins/woocommerce/includes/class-wc-checkout.php#L415). Then the `class-wc-order.php` has [the setter](https://github.com/Automattic/woocommerce.com/blob/trunk/plugins/woocommerce/includes/class-wc-order.php#L1265-L1270) which [sanitizes the email value](https://github.com/Automattic/woocommerce.com/blob/trunk/plugins/woocommerce/includes/class-wc-order.php#L1269C48-L1269C72).

Reply:
Yeah, it make sense that this email is already sanitized when it's set. I think you're safe to use the email as is here. 

>Eventually, this value will be still escaped [in WooCommerce Core](https://github.com/Automattic/woocommerce.com/blob/trunk/plugins/woocommerce/templates/myaccount/payment-methods.php#L51), so my escaping here might be redundant

I agree here as well. It would be escaped twice which isn't a problem, but even without escaping here, we're following the escape late principle and would still be safe IMO. So I'd say it's up to you whether you want to escape here or just pass the email through.

Reply:
Yep, glad we're aligned now. I'll keep the escaping here and thus close the thread without new commits. Thanks for the help!